### PR TITLE
Fix static framework archive regression from #7187

### PR DESCRIPTION
**Pull request by paulb777**
rsync does not automatically follow file links and was silently failing to copy the fat archive

Thanks to @noppefoxwolf and @roxiemobile for trouble-shooting and raising the issue in #7207!

### CODE REVIEW

File: lib/cocoapods/installer/xcode/pods_project_generator/pod_target_installer.rb [hunk 1]
```diff
@@ -471,7 +471,7 @@ def create_build_phase_to_move_static_framework_archive
             build_phase = native_target.new_shell_script_build_phase('Setup Static Framework')
             build_phase.shell_script = <<-eos.strip_heredoc
           mkdir -p "${BUILT_PRODUCTS_DIR}/${PRODUCT_NAME}.framework/Modules"
-          rsync -t "${BUILT_PRODUCTS_DIR}/lib${PRODUCT_NAME}.a" "${BUILT_PRODUCTS_DIR}/${PRODUCT_NAME}.framework/${PRODUCT_NAME}"
+          rsync -tL "${BUILT_PRODUCTS_DIR}/lib${PRODUCT_NAME}.a" "${BUILT_PRODUCTS_DIR}/${PRODUCT_NAME}.framework/${PRODUCT_NAME}"
```
Comment (dnkoutso) [inline lib/cocoapods/installer/xcode/pods_project_generator/pod_target_installer.rb]: for posterity can you explain the flag? :)

Comment (dnkoutso) [reply lib/cocoapods/installer/xcode/pods_project_generator/pod_target_installer.rb]: Do any other the `rsync` calls need it?

Comment (paulb777) [reply lib/cocoapods/installer/xcode/pods_project_generator/pod_target_installer.rb]: I verified that everything else gets copied. I believe it's related to the construction of the fat library archive that Xcode is using a file link for it.

Comment (dnkoutso) [reply lib/cocoapods/installer/xcode/pods_project_generator/pod_target_installer.rb]: great.

**Comment by dnkoutso** [comment]
Merge once green!

**Pull request merged**